Modules/ITS/src/ITSChipStatusTask.cxx

```diff
@@ -0,0 +1,178 @@
+      iLayerBegin = 5;
+      iLayerEnd = 6;
+    }
+    const int ChipBoundary[8] = { 0, 108, 252, 432, 3120 - 432, 6480 - 432, 14712 - 6480, 24120 - 6480 }; // needed for drawing labels and lines for histogram
+    const int StaveBoundary[NLayer] = { 0, 12, 28, 0, 24, 0, 42 };
+    for (Int_t iLayer = iLayerBegin; iLayer <= iLayerEnd; iLayer++) {
+
+      TLatex* msg = new TLatex(5, ChipBoundary[iLayer + 1] - 15 * (1 + i * 30), Form("#bf{L%d}", iLayer));
+      msg->SetTextSize(20);
+      msg->SetTextFont(43);
+      DeadChips[i]->GetListOfFunctions()->Add(msg);
+      if (iLayer < iLayerEnd) {
+        auto l = new TLine(0, ChipBoundary[iLayer + 1] - 0.5, nQCCycleToMonitor, ChipBoundary[iLayer + 1] - 0.5);
+        DeadChips[i]->GetListOfFunctions()->Add(l);
+      }
+    }
+    int nChipsPerStave = i == 0 ? 9 : i == 1 ? 112
+                                             : 196;
+    for (int iy = 0; iy < DeadChips[i]->GetNbinsY(); iy++) {
+      if (iy % nChipsPerStave == 0) {
+        auto l = new TLine(0, iy - 0.5, nQCCycleToMonitor, iy - 0.5);
+        l->SetLineColor(40);
+        l->SetLineWidth(1);
+
+        DeadChips[i]->GetListOfFunctions()->Add(l);
+
+        if (iy >= ChipBoundary[Layer_Draw + 1]) {
+          Layer_Draw++;
+        }
+        TLatex* msg = new TLatex(-3, iy, Form("%d", (int)(iy) / nChipsPerStave - StaveBoundary[Layer_Draw]));
+        msg->SetTextSize(12);
+        msg->SetTextFont(43);
+        DeadChips[i]->GetListOfFunctions()->Add(msg);
+      }
+    }
+    Layer_Draw++;
+
+    DeadChips[i]->GetYaxis()->SetLabelSize(0);
+    DeadChips[i]->GetYaxis()->SetAxisColor(0);
+    getObjectsManager()->startPublishing(DeadChips[i]);
+  }
+}
+
+void ITSChipStatusTask::setAxisTitle(TH1* object, const char* xTitle, const char* yTitle)
+{
+  object->GetXaxis()->SetTitle(xTitle);
+  object->GetYaxis()->SetTitle(yTitle);
+}
+
+void ITSChipStatusTask::startOfActivity(const Activity& activity)
+{
+  ILOG(Debug, Devel) << "startOfActivity : " << activity.mId << ENDM;
+}
+
+void ITSChipStatusTask::startOfCycle()
+{
+  ILOG(Debug, Devel) << "startOfCycle" << ENDM;
+
+  for (int i = 0; i < 3; i++) {
+    CurrentDeadChips[i].clear();
+    CurrentDeadChips[i].resize(ChipsBoundaryBarrels[i + 1] - ChipsBoundaryBarrels[i], 0);
+  }
+}
+
+void ITSChipStatusTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  auto aliveChips = ctx.inputs().get<gsl::span<char>>("chipstatus");
+
+  int id = 0;
+  for (auto chipID : aliveChips) {
+    int iBarrel = id < ChipsBoundaryBarrels[1] ? 0 : id < ChipsBoundaryBarrels[2] ? 1
+                                                                                  : 2;
+    if ((int)chipID != 1) {
+      int ChipID_fill = id - ChipsBoundaryBarrels[iBarrel];
+      if ((ChipID_fill < 0) || (ChipID_fill >= CurrentDeadChips[iBarrel].size())) {
+        ILOG(Warning, Devel) << " prolematic ID for chip: " << ChipID_fill << ENDM;
+      } else {
+        CurrentDeadChips[iBarrel][ChipID_fill]++;
```

Comment:
Hi @IsakovAD , do I well understand that this CurrentDeadChips is counting the number of TFs in the cycle for which the ChipID_fill was dead.. and this information will go to the TH2?
May I suggest to normalize it by the number of TFs (=monitorData cycles) processed in the QC cycle ? This would give absolute and understandable value non depending on the settings and on fluctuactions.

Another possibility is to use some absolute time unit on the x axis, e.g. orbit number instead of QC cycle.

Reply:
Unfortunately, I can't use adaptive x-axis. 



But to show fraction, now we have TH2Fraction class which stores number of TFs per QCCycle in denominator